util/wrapper/wrapper.go

```diff
@@ -126,3 +130,51 @@ func TraceHandler(t trace.Tracer) server.HandlerWrapper {
 		}
 	}
 }
+
+// AuthHandler wraps a server handler to perform auth
+func AuthHandler(a auth.Auth) server.HandlerWrapper {
+	return func(h server.HandlerFunc) server.HandlerFunc {
+		return func(ctx context.Context, req server.Request, rsp interface{}) error {
+			// Extract endpoint and remove service name prefix
+			// (e.g. Platform.ListServices => ListServices)
+			var endpoint string
+			if ec := strings.Split(req.Endpoint(), "."); len(ec) == 2 {
+				endpoint = ec[1]
+			}
+
+			// Check for endpoints excluded from auth. If the endpoint
+			// matches, execute the handler and return
+			for _, e := range a.Options().Excludes {
+				if e == endpoint {
+					return h(ctx, req, rsp)
+				}
+			}
+
+			// Extract the Authorization header
+			md, ok := metadata.FromContext(ctx)
+			if !ok {
+				md = metadata.Metadata{}
+			}
+			authHeader := md["Authorization"]
```

Comment:
call metadata.Get(ctx, "Authorization") as if this passes through http and http2 it will go through various case formats. Either that or metadata.FromContext needs to camel case keys.